napari_aicsimageio/core.py

```diff
@@ -41,6 +63,77 @@ def _get_full_image_data(img: AICSImage, in_memory: bool) -> Optional[xr.DataArr
     return None
 
 
+# Function to handle multi-scene files.
+def _get_scenes(img: AICSImage, in_memory: bool) -> Optional[xr.DataArray]:
+    # Create the list widget and populate with the scenes in the file
+    list_widget = QListWidget()
+    for scene in img.scenes:
+        list_widget.addItem(scene)
+    viewer = _get_viewer()
+    viewer.window.add_dock_widget([list_widget], area="right", name="Scene Selector")
+
+    # Function to create image layer from a scene selected in the list widget
+    def open_scene(item):
+        scene = item.text()
+        img.set_scene(scene)
```

Comment:
So I have a safety measure request and you haven't noticed this behavior because you are using LIFs where we construct the scene name.

Other file formats can have duplicate scene names and as such when calling `set_scene` it may fail to set the appropriate / desired scene. Can you change to the following:

```python
for i, scene in enumerate(img.scenes):
    list_widget.addItem(f"{i} -- {scene}")

# ... viewer stuff

def open_scene(item):
    scene_text = item.text()
    scene_index = int(scene_text.split(" -- ")[0])
    img.set_set_scene(scene)
```

Reply:
This _should_ be a safe way for ensuring the scene is selected by integer index rather than name.

Reply:
OMG! I tried multiple times to get the index in the Item name! I wanted that to be able to easy pull groups of scenes programmatically like you showed me before! So clever! I never thought to just make a new name and then split! Wow!

Reply:
```
def open_scene(item):
    scene_text = item.text()
    scene_index = int(scene_text.split(" -- ")[0])
    img.set_set_scene(scene)
```
In this last line, we'd want it to be:
    `img.set_set_scene(scene_index)`
right? To deal with duplicate names (which certainly can happen, especially now without the `readlif` branch).

Reply:
Sorry! Correct. You want to provide the `scene_index` (the integer) to the `set_scene` method.